### PR TITLE
added public repo support

### PR DESCRIPTION
**Pull request by GodReaper**
Screenshot 2024-10-30 at 4 58 02 PM
Screenshot 2024-10-30 at 4 58 14 PM

### CODE REVIEW

File: app/modules/github/github_service.py [hunk 1]
```diff
@@ -254,7 +254,36 @@ async def get_repos_for_user(self, user_id: str):
             raise HTTPException(
                 status_code=500, detail=f"Failed to fetch repositories: {str(e)}"
             )
-
+    async def get_combined_user_repos(self, user_id: str):
+        parsed_repos = await self.project_manager.list_projects(user_id)
```
Comment (dhirenmathur) [inline app/modules/github/github_service.py]: Fetch unique repo name

File: app/modules/github/github_service.py [hunk 2]
```diff
@@ -254,7 +254,36 @@ async def get_repos_for_user(self, user_id: str):
...
+        project_list = [
+            {
+                "id": project["id"],
+                "name": project["repo_name"].split('/')[-1],
+                "full_name": project["repo_name"],
+                "private": False,
```
Comment (dhirenmathur) [inline app/modules/github/github_service.py:273]: do not mark all as public

File: app/modules/github/github_service.py [hunk 3]
```diff
@@ -254,7 +254,36 @@ async def get_repos_for_user(self, user_id: str):
...
+                "url": f"https://github.com/{project['repo_name']}",
+                "owner": project["repo_name"].split('/')[0]
+            }
+            for project in parsed_repos
+        ]
+        user_repo_response = await self.get_repos_for_user(user_id)
```
Comment (dhirenmathur) [inline app/modules/github/github_service.py:279]: filter all the github response repos from unique list from db, then mark all the db projects as public

**Review by dhirenmathur — DISMISSED** [review]

**Comment by vineetshar** [comment]
@GodReaper not getting enough context from screenshots, how exactly do I add public repos? can I have detailed screenshots please - including confirmation that they get parsed

**Review by vineetshar — APPROVED** [review]
LGTM

**Pull request merged**